### ISSUE TITLE
Make setup process easier

### ISSUE THREAD

**Issue by eduardoboucas** [speedtracker/speedtracker#27]
Currently, the process of setting up a new dashboard can trip up some people. After obtaining a WebPageTest API key, users need to:

1. Add the bot to their repository
1. Use the connect tool to prompt the bot accept the invitation
1. Use the encrypt tool to create an encrypted version of a chosen passkey
1. Add the encrypted passkey to the config file
1. Use the encrypt tool to create an encrypted version of the WPT API key with the previously created passkey
1. Add the encrypted version of the WPT API key to the config file
1. Edit the default profile file and edit the name of the property and the URL

I'd like to have a UI tool that displays a form including:

- Repository URL
- Chosen passkey
- WPT API key
- Name of the default property
- URL of the default property

By filling in these details, the API would then take care of encrypting everything it needs to and write both the config file (`staticman.yml`)  and the first profile (`_profiles/default.html`).

This endpoint will be open, as in people will be able to run this against their newly created repository without providing any sort of authentication. For this reason, we need to make sure this procedure can be ran only once, when the repository hasn't yet been modified by its owner (not sure how to determine this).

To achieve all this, I see two separate components:

1. A new API endpoint (or an extension of the `/connect` endpoint to optionally accept the extra parameters)
1. A new page on speedtracker.org containing a form and all the necessary logic to create and send the request to the above-mentioned endpoint. Instead of being a new page, this could also be an extension to the Connect page, adding new elements to the existing form.

I can handle most (if not all) of part **1**, but I could use a hand with part **2**, the UI for it. The existing markup and logic for the Connect, Encrypt and Run Test pages is a bit messy. I can see the need for more robust, responsive UI controls, perhaps using something like React/Preact.

**Comment by eduardoboucas** [comment]
This could even encompass running the first test as well.

**Comment by avadhutp** [comment]
How about just a script? The script will be triggered by the end user in his/her cloned repo. And it will change files, etc. In the end, it will simply push the changes + make the first connect.

**Comment by eduardoboucas** [comment]
Yeah, that makes sense. It can also fire the first test, which would then trigger any scheduled tests.

**Comment by avadhutp** [comment]
@eduardoboucas Cool, I will take a jab at this.

**Comment by eduardoboucas** [comment]
Awesome, thanks! There might be things that I haven't fully documented properly, so if you need assistance with anything I'm happy to help.

**Comment by avadhutp** [comment]
@eduardoboucas Cool. I will mark you on WIP pull request when I get to it this evening. :-)
